### ISSUE TITLE
Confluence macro

### ISSUE THREAD

**Issue by OLibutzki** [WPS/egon.io#59]
The Domain Story Modeler is a very good tool to capture and refine the results of a modelling workshop.

It would be cool, if you could provide a Confluence macro to integrate the editor in Confluence. The main benefit is that you can view and edit the model in Confluence directly. So there is no need to use an external tool, upload the dst file and provide an exported image to include it in Confluence for documentation purposes.

As the modeler is based on bpmn.io it might not be that hard to realize this macro as there is a working [Confluence macro for BPMN](https://marketplace.atlassian.com/apps/1215543/bpmn-modeler-free?hosting=server&tab=overview).

We use Camunda BPMN a lot and viewing, discussing and editing the models in Confluence in a collaborative way is a huge benefit. Maybe it's an option for the Domain Sotry Modeler as well.

Kind regards
Oliver

**Comment by hofstef** [comment]
Definitely worth a look. I decided to label this an idea. To turn it into a implementable feature, we need to think about support, availabilty, compatibility etc.

**Comment by OLibutzki** [comment]
Btw, I stumbled upon this tweet about integrating bpmn.io diagrams in GitHub: https://twitter.com/lutki95/status/1162356190711615496

Might be interesting as well...

**Comment by hofstef** [comment]
we are taking a serious look at this feature to see if it is feasable

**Comment by hofstef** [comment]
Tl;dr: We are putting this idea on hold.

Long version:

The main goal of an integration would be to get rid of this cycle: download from confluence/upload to modeler/edit in modeler/download from modeler/upload in confluence. Instead, you would edit (and re-replay) .dst-files “inline”, like for example draw.io does.

That means a “shallow” integration (wrapping the modeler in an iFrame macro) will not be useful. Hence, we would need to develop a proper confluence plugin. We did only scratch the surface of how to do that but here are some things that we would need to consider:
-	Handling of custom icons would need to change so that pallet shows the correct icons
-	Changes to the UI that result from different handling saving/importing/exporting models
-	Enable replay on restricted pages
-	Ensure compatibility with different confluence versions
-	Offer professional support

All of this is doable but would take considerable effort. That’s why I am putting this on hold until we figured out some sort of business model that enables us to provide enterprise-features on top of the free modeler.

**Comment by OLibutzki** [comment]
Ok, for the BPMN Confluence Macro viadee also provides a free and an enterprise edition. Maybe you can get some inspiration concerning the business model by looking at there enterprise features.
[Free Modeler](https://marketplace.atlassian.com/apps/1215543/bpmn-modeler-free?hosting=server&tab=overview)
[Enterprise Modeler](https://marketplace.atlassian.com/apps/1219284/bpmn-modeler-enterprise?hosting=server&tab=overview)
